core/flyout_navigator.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import type {IFlyout} from './interfaces/i_flyout.js';
+import {FlyoutButtonNavigationPolicy} from './keyboard_nav/flyout_button_navigation_policy.js';
+import {FlyoutNavigationPolicy} from './keyboard_nav/flyout_navigation_policy.js';
+import {FlyoutSeparatorNavigationPolicy} from './keyboard_nav/flyout_separator_navigation_policy.js';
+import {Navigator} from './navigator.js';
+
+export class FlyoutNavigator extends Navigator {
+  constructor(flyout: IFlyout) {
+    super();
+    this.rules.push(
+      new FlyoutButtonNavigationPolicy(),
+      new FlyoutSeparatorNavigationPolicy(),
+    );
```

Comment:
Would it perhaps be cleaner to introduce something like a protected init method that takes the list of rules to initialize? Then this navigator could fully override the behavior and very specifically indicate which rules should be present (rather than only being able to add onto them, or needing to overwrite the list).

This also would avoid needing to over-expose the field.

Reply:
Attempted this, but TS/JS initialization rules make it complicated. Punting for now.